File: cslacs10/CSLA.Test/Grandchildren.vb
<Serializable()> _
Public Class Grandchildren
  Inherits BusinessCollectionBase

  Default Public ReadOnly Property Item(ByVal index As Integer) As Grandchild
    Get
      Return CType(list(index), Grandchild)
    End Get
  End Property

  Public Sub Add(ByVal Data As String)
    list.Add(Grandchild.NewGrandChild(Data))
  End Sub

  Public Sub Remove(ByVal child As Grandchild)
    list.Remove(child)
  End Sub

  Friend Shared Function NewGrandChildren() As Grandchildren
    Return New Grandchildren
  End Function

  Friend Shared Function GetGrandChildren(ByVal dr As IDataReader) As Grandchildren
    ' TODO: load child data
  End Function

  Friend Sub Update(ByVal tr As IDbTransaction)
    Dim child As Grandchild

    For Each child In list
      child.Update(tr)
    Next
  End Sub

  Private Sub New()
    ' prevent direct creation
    MarkAsChild()
  End Sub

End Class

File: Csla20Test/Csla.Test/Basic/GrandChildren.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Data;

namespace Csla.Test.Basic
{
    [Serializable()]
#if csla20cs
    public class GrandChildren : BusinessListBase<GrandChild>
    {
        public void Add(string data)
        {
            this.Add(GrandChild.NewGrandChild(data));
        }

        internal static GrandChildren NewGrandChildren()
        {
            return new GrandChildren();
        }

        internal static GrandChildren GetGrandChildren(IDataReader dr)
        {
            //todo: load child data
            return null;
        }

        internal void Update(IDbTransaction tr)
        {
            foreach (GrandChild child in this)
            {
                child.Update(tr);
            }
        }

        private GrandChildren()
        {
            //prevent direct creation
            MarkAsChild();
        }
    }
#endif
    #if csla20vb
    public class GrandChildren : BusinessListBase<GrandChildren, GrandChild>
    {
        public void Add(string data)
        {
            this.Add(GrandChild.NewGrandChild(data));
        }

        internal static GrandChildren NewGrandChildren()
        {
            return new GrandChildren();
        }

        internal static GrandChildren GetGrandChildren(IDataReader dr)
        {
            //todo: load child data
            return null;
        }

        internal void Update(IDbTransaction tr)
        {
            foreach (GrandChild child in this)
            {
                child.Update(tr);
            }
        }

        private GrandChildren()
        {
            //prevent direct creation
            MarkAsChild();
        }
    }
#endif
}
